.github/workflows/deployment.yml

```diff
@@ -38,3 +39,53 @@ jobs:
           prerelease: true
           title: "Main"
           files: dist.zip
+
+  deploy_npm:
+    name: "Deploy NPM package"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+          registry-url: "https://npm.pkg.github.com"
+          scope: "@${{ github.repository_owner }}"
+
+      - run: npm ci
+
+      - name: Building
+        run: npm run css:build
+
+      - name: Bundling assets
+        run: ./bundle.sh
```

Comment:
Could be nice if these steps could be defined in one place, instead of both in "deploy_release" and "deploy_npm".
Not crucial, just a thought.

Reply:
I agree. Unfortunately I do not think GitHub has a good way to do this.

One way to approach this would be to build and bundle in one job and then fan out to different kind of releases. You can either use cache or artifacts to share the bundle. Artifacts are so slow that is frequently faster just to install packages etc. again. Caching should - in theory at least - be ephemeral and thus do not guarantee availability between jobs.

We could also use a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) which can be reused across jobs. However I fear that would be more trouble that it is worth at this point in time.

All in all I think the best approach for now it to keep things as is.

Reply:
👍